### ISSUE TITLE
darwin: python multiprocessing broken with sandboxed build

### ISSUE THREAD

**Issue by datakurre** [NixOS/nixpkgs#10092]
I'm on Darwin (OSX 10.10.5) with stdenvDarwinPure and using sandboxed (chroot = true) build for everything to avoid random impurity issues here and there. One of the side-effects seem to be that my Python multiprocessing module is partly broken. E.g. https://github.com/NixOS/nixpkgs/issues/10091 and

```
>>> import multiprocessing
>>> multiprocessing.RLock()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/nix/store/nr4yvydamabs72m4yzr8xzjniqfk9vjj-python-2.7.10/lib/python2.7/multiprocessing/__init__.py", line 182, in RLock
    from multiprocessing.synchronize import RLock
  File "/nix/store/nr4yvydamabs72m4yzr8xzjniqfk9vjj-python-2.7.10/lib/python2.7/multiprocessing/synchronize.py", line 59, in <module>
    " function, see issue 3770.")
ImportError: This platform lacks a functioning sem_open implementation, therefore, the required synchronization primitives needed will not function, see issue 3770.
```

**Comment by datakurre** [comment]
It might be enough to force HAVE_SEM_OPEN in configure, somehow.

**Comment by datakurre** [comment]
Fixed in Nix 1.11 with fixes in darwin sandbox.
